components/speedreader/rust/lib/src/readability/src/scorer.rs

```diff
@@ -108,10 +110,21 @@ impl PartialEq for TopCandidate {
 
 impl Eq for TopCandidate {}
 
-#[derive(Default)]
-pub struct Title {
+#[derive(Default, Debug)]
+pub struct Meta {
     pub title: String,
-    pub is_meta: bool,
+    pub author: String,
+    pub description: String,
+    pub charset: String,
+}
+
+impl Meta {
+    pub fn new() -> Meta {
+        Meta {
+            charset: "utf-8".to_string(),
+            ..Default::default()
+        }
+    }
 }
```

Comment:
If you're only using the `Default` derive for the `new` function, it's probably clearer to just create a custom `Default` implementation.

```rust
impl Default for Meta {
    fn default() -> Self {
        Self {
            title: Default::default(),
            author: Default::default(),
            description: Default::default(),
            charset: "utf-8".to_string(),
        }
    }
}
```

I do see you're using this in a test-only `preprocess` method; I didn't look too closely if it's needed differently there but it'd be good to make that difference explicit if that's the case.

Reply:
Also, according to the [MDN docs for `meta charset`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta)
> This attribute declares the document's character encoding. If the attribute is present, its value must be an ASCII case-insensitive match for the string "`utf-8`", because UTF-8 is the only valid encoding for HTML5 documents.

If having this information is even useful, perhaps it's worth making into an enum? Something like

```rust
enum MetaCharset {
    Unknown,
    Utf8,
}
```

with a constructor that matches on a `&str` argument.

Reply:
There are some others that we would need to support. https://www.w3schools.com/charsets/

I thought of doing the enum too, but speedreader isn't going to validate the charsets anyway. there seems to be a bunch of revisions to ISO-8859, so i wouldn't even know how many to include.